server/src/main/java/org/apache/druid/client/DirectDruidClient.java

```diff
@@ -159,14 +158,15 @@ public int getNumOpenConnections()
 
     Pair<JavaType, JavaType> types = typesMap.get(query.getClass());
     if (types == null) {
```

Comment:
It could improve concurrency, see https://github.com/apache/incubator-druid/pull/4397#discussion_r122067079. I'll add a comment.

Reply:
Got it.

Reply:
Hmm, `computeIfAbsent()` will call `get()` again, so I think we shouldn't use `computeIfAbsent()` here. Just revert this modification.

Reply:
Hmm, `computeIfAbsent()` may call `get()` again(ConcurrentHashMap doesn't, ConcurrentSkipListMap does), so I think we shouldn't use `computeIfAbsent()` here, using `put()` is ok.

Reply:
If two queries of the same new type are run in parallel, there could be a race between them. Maybe it could be tolerated here because computation (body of the lambda) could be run in parallel and re-run for the same type with no harm, but `computeIfAbsent()` is clearer.